### PR TITLE
prevent filename being None in Cell

### PR DESCRIPTION
**Pull request by darshanmandge**
To prevent `filename` being `None` in `Cell` class

### CODE REVIEW

File: bluepyefe/cell.py [hunk 1]
```diff
@@ -60,12 +60,18 @@ def reader(self, config_data, recording_reader=None):
...
-
-        if "v_file" in config_data:
-            filename = config_data["v_file"]
-        # if both present: use filepath. e.g. for some nwb that 'contain' igor files
+        # if both present: use filepath.
+        # e.g. for some nwb that 'contain' igor files
+        filename = None
         if "filepath" in config_data:
             filename = config_data["filepath"]
+        if "v_file" in config_data and filename is None:
```
[the remark this answers is not in the document]
Comment (darshanmandge) [reply bluepyefe/cell.py:68]: If I use `elif`, the `filename` may have `None` which we don't want as a default `None' is given for `config_data["filepath"]` by bluepyemodel when using `igor_reader`

File: bluepyefe/cell.py [hunk 2]
```diff
@@ -60,12 +60,18 @@ def reader(self, config_data, recording_reader=None):
...
+            filename = config_data["v_file"]
+
+        if filename is None:
+            raise Exception(
+                "No 'filename' provided in the metadata for the recording."
```
[the remark this answers is not in the document]
Comment (darshanmandge) [reply bluepyefe/cell.py:73]: Does this sound right?: 
"No 'filepath' or 'v_file' provided in the metadata for the recording."

[the remark this answers is not in the document]
Comment (AurelienJaquier) [reply bluepyefe/cell.py]: Yes, sounds good. Can you add or 'v_file' then?

File: bluepyefe/cell.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (AurelienJaquier) [reply bluepyefe/cell.py:68]: Oh, if config_data["filepath"] is None. Ok, I see, it's good like this then

File: bluepyefe/cell.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (darshanmandge) [reply bluepyefe/cell.py]: Updated. I also changed the local variable name from `filename` to `filepath` as it much accurate. filename may not have the extension which we check below

**Review by ilkilic — APPROVED** [review]
Thank you @darshanmandge !

**Pull request merged**